File: tests/machine/x/php/dataset_sort_take_limit.php
<?php
$products = [
    ["name" => "Laptop", "price" => 1500],
    ["name" => "Smartphone", "price" => 900],
    ["name" => "Tablet", "price" => 600],
    ["name" => "Monitor", "price" => 300],
    ["name" => "Keyboard", "price" => 100],
    ["name" => "Mouse", "price" => 50],
    ["name" => "Headphones", "price" => 200]
];
$expensive = (function() use ($products) {
    $result = [];
    foreach ($products as $p) {
        $result[] = [-$p['price'], $p];
    }
    usort($result, function($a, $b) { return $a[0] <=> $b[0]; });
    $result = array_map(fn($r) => $r[1], $result);
    $result = array_slice($result, 1, 3);
    return $result;
})();
echo "--- Top products (excluding most expensive) ---", PHP_EOL;
foreach ($expensive as $item) {
    var_dump($item['name'], "costs $", $item['price']);
}
?>

File: tests/aster/x/fs/dataset_sort_take_limit.fs
// Generated 2025-07-21 18:37 +0700

type Anon1 = {
    name: string
    price: int
}


type Anon2 = {
    name: string
    price: int
}


let products: Anon2 list = [{ name = "Laptop"; price = 1500 }; { name = "Smartphone"; price = 900 }; { name = "Tablet"; price = 600 }; { name = "Monitor"; price = 300 }; { name = "Keyboard"; price = 100 }; { name = "Mouse"; price = 50 }; { name = "Headphones"; price = 200 }]

let expensive = [for p in List.take 3 (List.skip 1 (List.sortBy (fun p -> (-(p.price))) products)) do

]

printfn "%s" (string "--- Top products (excluding most expensive) ---")

item

expensive

printfn "%s"

(String.concat " " [string (item.name); string "costs $"; string (item.price)])
